src/frontend/src/components/OECI/index.tsx

```diff
@@ -0,0 +1,153 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { AppState } from '../../redux/store';
+import { logIn } from '../../redux/system/actions';
+import { SystemState } from '../../redux/system/types';
+import history from '../History';
+
+interface Props {
+  system: SystemState;
+  logIn: typeof logIn;
+}
+interface State {
+  userId: string;
+  password: string;
+  userIdHasInput: boolean;
+  passwordHasInput: boolean;
+  invalidCredentials: null | boolean;
+  missingInputs: null | boolean;
+}
+
+class OECIlogin extends React.Component<Props, State> {
+  state: State = {
+    userId: '',
+    password: '',
+    userIdHasInput: false,
+    passwordHasInput: false,
+    invalidCredentials: null,
+    missingInputs: null
+  };
+
+  handleChange = (e: React.BaseSyntheticEvent) => {
+    // See https://github.com/DefinitelyTyped/DefinitelyTyped/issues/26635 for why we're
+    // using the "any" type.
+    this.setState<any>({
+      [e.target.id]: e.target.value
+    });
+  };
+
+  handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    this.validateForm();
+  };
+
+  validateForm = () => {
+    this.setState({
+      userIdHasInput: this.state.userId.trim().length === 0
+    });
+    this.setState({
+      passwordHasInput: this.state.password.trim().length === 0
+    });
+    this.setState(
+      {
+        missingInputs:
+          this.state.userId.trim().length === 0 ||
+          this.state.password.trim().length === 0
+      },
+      () =>
+        this.state.missingInputs === false
+          ? history.push('/recordsearch')
+          : null // Using setState(updater, callback) because setState doesn't immediately update component
+    );
```

Comment:
I realize the backend logic isn't in place so it's not possible to totally anticipate how this code should look, but we should not mutate global state (the history) as a side effect of validation. The history should only be updated if the login attempt was successful and the user is actually changing pages in response to that. So the logic in this callback should be moved to `handleSubmit` or removed for now.

Reply:
I've moved the logic to `handleSubmit` and changed it utilize `<Redirect to='#'>`